File: ts-sdk/src/sdk/models/shared/workflowoccurrence.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { StageStatus } from "./stagestatus";


export class WorkflowOccurrence extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=createdAt" })
  createdAt: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=statuses", elemType: StageStatus })
  statuses: StageStatus[];

  @SpeakeasyMetadata({ data: "json, name=updatedAt" })
  updatedAt: Date;

  @SpeakeasyMetadata({ data: "json, name=workflowID" })
  workflowID: string;
}

File: java-sdk/lib/src/main/java/dev/formance/javasdk/models/shared/WorkflowOccurrence.java
package dev.formance.javasdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import dev.formance.javasdk.utils.DateTimeDeserializer;
import dev.formance.javasdk.utils.DateTimeSerializer;
import java.time.OffsetDateTime;

public class WorkflowOccurrence {
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("createdAt")
    public OffsetDateTime createdAt;
    public WorkflowOccurrence withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    
    @JsonProperty("id")
    public String id;
    public WorkflowOccurrence withId(String id) {
        this.id = id;
        return this;
    }
    
    @JsonProperty("statuses")
    public StageStatus[] statuses;
    public WorkflowOccurrence withStatuses(StageStatus[] statuses) {
        this.statuses = statuses;
        return this;
    }
    
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("updatedAt")
    public OffsetDateTime updatedAt;
    public WorkflowOccurrence withUpdatedAt(OffsetDateTime updatedAt) {
        this.updatedAt = updatedAt;
        return this;
    }
    
    @JsonProperty("workflowID")
    public String workflowID;
    public WorkflowOccurrence withWorkflowID(String workflowID) {
        this.workflowID = workflowID;
        return this;
    }
    
}
